ui/src/main/java/org/jtaccuino/core/ui/controls/InputControl.java

```diff
@@ -115,49 +110,24 @@ private void setup() {
         input.getStyleClass().add(type.styleClassPrefix + "-editor");
         input.setAutoSave(true);
 
-        input.documentProperty().subscribe(d
-                -> Platform.runLater(()
-                        -> recalculateRTA(Math.max(MIN_WIDTH, getWidth()))));
-        input.heightProperty().subscribe((h0, h) -> {
-            double newHeight = h.doubleValue() + 2 * padding;
-            setMinHeight(newHeight);
-            setPrefHeight(newHeight);
-            setMaxHeight(newHeight);
-            requestLayout();
+        input.fullHeightProperty().subscribe((h0, h) -> {
+            recalculateRTA(getWidth());
         });
         widthProperty().subscribe(w -> recalculateRTA(w.doubleValue()));
     }
 
     private void recalculateRTA(double width) {
         input.setPrefWidth(width - 2);
-        double textAreaHeight = computeRTAPrefHeight(input.getWidth()) + inputPadding;
+        double textAreaHeight = input.getFullHeight() + inputPadding;
         input.setMinHeight(textAreaHeight);
         input.setPrefHeight(textAreaHeight);
         input.setMaxHeight(textAreaHeight);
         input.requestLayout();
+        
+        double newHeight = textAreaHeight + 2;
+        setMinHeight(newHeight);
+        setPrefHeight(newHeight);
+        setMaxHeight(newHeight);    
```

Comment:
This shouldn't be here: for markdown, the `InputControl` (`MarkdownControl`) has two RTAs, `input` and `mdRenderArea`, and on Markdown mode, the former is not visible/managed, so it shouldn't be used to define the height of the control

Reply:
anything on this?

Reply:
Yes code was updated to better handle the different states of rendering (source vs. preview). This is probably more complex, but feels more safe.